[14.0][IMP] product_import: currency can be implicit

florentx opened the pull request:
When currency was not on each catalogue line, there was tons of warnings on importation.

**[ EDIT ]**

Rebased on top of refactor PR:
- #1076 
- #1077

florentx committed:
[IMP] product_import: optionally set company on product

florentx committed:
[IMP] product_import: find product by internal reference

florentx committed:
[IMP] product_import: currency can be implicit

twalter-c2c reviewed (commented):
Checks failed on `NameError: name 'import_company_id' is not defined`. Should it be `product_company_id` instead maybe?

florentx commented:
Fixed.

simahawk commented:
A test would be nice... as usual :wink:

florentx committed:
[IMP] product_import: Reorganize for inheritance to import by batch

florentx committed:
[REF] product_import: do not use the env.context to configure import

florentx committed:
[FIX] product_import: use update() instead of write()

It is needed in order to have a safe recomputation of fields of other models.
There was an issue in a third-party add-on which has a field recomputed
when a new 'product.supplierinfo' is added to 'seller_ids' of the product.
